code/modules/mining/machine_vending.dm

```diff
@@ -245,6 +245,11 @@
 		qdel(src)
 
 
+/obj/machinery/mineral/equipment_vendor/AltClick(mob/user)
+	inserted_id.forceMove(loc)
+	inserted_id.verb_pickup()
+	inserted_id = null
```

Comment:
```suggestion
	if(!user.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
		return
	inserted_id.forceMove(loc)
	inserted_id.verb_pickup()
	inserted_id = null
```
Still assuming you meant silicons to not be able to use it far away (edit if unintentional but otherwise this check is needed to let people not use it from a distance)

Reply:
Silicons shouldnt be messing with that